tiledb/sm/array/array.cc

```diff
@@ -1560,5 +1560,157 @@ void Array::set_serialized_array_open() {
   is_open_ = true;
 }
 
+void Array::get_max_tile_size(
+    tiledb_fragment_max_tile_sizes_t* max_tile_sizes) {
+  assert(max_tile_sizes);
+
+  if (remote_) {
+    auto rest_client = storage_manager_->rest_client();
+    if (rest_client == nullptr) {
+      throw StatusException(
+          Status_Error("Failed to obtain maximum tile size; remote array with "
+                       "no REST client."));
+    }
+
+    throw_if_not_ok(rest_client->get_array_maximum_tile_size_from_rest(
+        array_uri_, &max_tile_sizes->max_in_memory_tile_size));
+    return;
+  }
+
+  // Check that array is open
+  if (!is_open()) {
+    throw StatusException(Status_ArrayError(
+        "[Array::array_get_fragment_tile_max_size] Array is closed"));
+  }
+
+  if (query_type_ != QueryType::READ) {
+    throw StatusException(Status_ArrayError(
+        "[Array::array_get_fragment_tile_max_size] needs QueryType::READ"));
+  }
+
+  memset(max_tile_sizes, 0, sizeof(*max_tile_sizes));
```

Comment:
We do this already below so I think we can remove this one.

Reply:
constructor was added, max_tile_sizes usage in method somewhat changed in general